### ISSUE TITLE
LoadDoctor Phase 2

### ISSUE THREAD

**Issue by Harmonickey** [LbKStudios/lbkstudios-main#143]
Of course, we still need to propose a time span (he doesn't really want it to last more than 2 weeks, which is fine, we'll just say 2 weeks) and a lump cost (which I think should just be $300 again based on the work that needs to be done).

Login with
CarryTransit
CarryTransit@1

Project Overview - Carrier.aspx:
~~1. Optimize the code and frontend to display the data as it's presented instead of the current behavior, where it spawns exactly 100 divs and then removes remaining divs as others are filled in by data.
    -- transform the 100-queries-for-100-items- into 1-query-that-gets-100-items~~
~~2. Analyze and optimize if possible, the "bid entry" mechanism for this webpage.~~
~~3. If possible, enhance the graphics to make it more engaging.  We can do whatever we want here as long as it makes sense.~~
~~4. Add a constraint that disallows "bid entry" when it's the midnight of DeliveryTime (make it actually some really far ahead future date, so that it is demoable).  There needs to be a countdown for the midnight constraint that the user can see.
      -- Countdown should have ##days ##hours ##minutes ##seconds and update realtime~~
      ~~-- Add 'disable' feature to the bid buttons to make it clear that the users cannot bid anymore.~~
5. ~~Add new datafields to the bidding platform if status = 0 from the database query.  The new fields will be MinDate-Time and MaxDate-Time, which ideally would be able to be selectable by a date object and time object. Using a tab for this to separate from the other bidding platform would be ideal.~~
  ~~-- the location where this happens is in GetTrucksData() which filters the database query on data that doesn't equal Status 2.  So therefore it gets all the Status == 1 and Status == 0.~~

**Comment by Saurutobi** [comment]
I uploaded the contract, can you re-write the last point(6) in there, it still doesn't make sense cuz I just copied the ticket

**Comment by Harmonickey** [comment]
Does it look good now?  I cleaned up the project attachment part.

**Comment by Saurutobi** [comment]
looks good, pdf is in there

**Comment by Harmonickey** [comment]
Okay, cool.  Sent it to him.  I've uploaded all the latest changes to the /dev branch.  We can just work from there and push back to /dev and then I'll pull on his server when we've finished.

**Comment by Harmonickey** [comment]
I have attempted at step 1.  The divs are now loaded dynamically and I also fixed a zooming bug (when zoomed and then zoomed back to 100%, most of the widths would not reset to their original size).  I sent Hami an email about the CarrierID and UserID thing, but otherwise it all looks pretty good.  I'll push what I got.

**Comment by Harmonickey** [comment]
I moved the data selection to be above the bid table since the bid table can increase and decrease in size dynamically, pushing the data selection.  Instead it would be better to have the data selection up top and let the table increase downwards below it.

In addition, the auction table is still next to the data selection.  We should think about adding in the countdown timer up there as well somehow.

**Comment by Harmonickey** [comment]
I added a loading bar that disables the page until everything is loaded.

**Comment by Harmonickey** [comment]
Enhancing more graphics, also added the countdown timer and constraint section in the code to disallow the user to bid if the current date is after the constraint date.  The user is met with an alert message if they try to do so.

**Comment by Harmonickey** [comment]
I changed up some code so that the frontend can parse out status==0 and status==1.  Now all that needs to happen is to create two tabs.  One tab for status==1 and one tab for status==0, then output the data into their respective tables.

**Comment by Harmonickey** [comment]
http://getbootstrap.com/components/#nav-tabs

**Comment by Saurutobi** [comment]
or you can use ASP Panels(which are awesome)

**Comment by Harmonickey** [comment]
Do you know how to use them?

**Comment by Saurutobi** [comment]
...yes

**Comment by Saurutobi** [comment]
jk, yes. it's really, stupid easy. It's like an html tag, and you can set them to blorp.visible=false/true in the c#

**Comment by Saurutobi** [comment]
I'm getting my shit setup, my github crashed

**Comment by Harmonickey** [comment]
css enhancements, template revision (CSS loaded in front of javascript to make sure loading is done in parallel), tab creation, and data insertion for status differences applied.

The only thing left is to figure out how to add in the MinTime and MaxTime in the status = 0 tab.

**Comment by Harmonickey** [comment]
And the loading bar now has more information when proccessing... "Loading Today's Data" or "Loading Data from 9/20/1992"  etc...

**Comment by Harmonickey** [comment]
My latest changes include moving the 'auction times' items into their own div, then the itinerary date selector in its own div, along with a shipper selector.  The shipper selector is a drop down list below the date selector and its values are the names of their respective databases (even though I agree really it should all be one database).  The auction time selector thing is now invisible unless there are auction times in the AuctionDiv.

**Comment by Harmonickey** [comment]
The status=0 section has minDate and maxDate datetime selectors.  I checked the functionality of everything and it looks good.  In addition, I changed the time selectors for the auction div.

**Comment by Harmonickey** [comment]
Added DB selector into code, so now the user can select between different shippers.  Fixed some random CSS things.  Changed the comment box to be a textarea.

**Comment by Harmonickey** [comment]
Phew, I did it.  I ended up needing to separate out the trucks into dictionaries because the trucks were being repeated many many times in the graphs.  With the dictionary, it made it so that they weren't repeated.  In addition, I had to fix the 'x' additions that are appended to every truck.  That was appending too many, so I just looped over each truck for each status and appended only one 'x' FleetSizeData to the truck data.

DUDE!!! It's FUCKING fast now.  Nice work.

We are done with this!
